src/shogun/mathematics/graph/Types.h

```diff
@@ -34,56 +34,81 @@ namespace shogun
 
 		class SHOGUN_ENGINE_EXPORT NumberType
 		{
-			public:
-				explicit NumberType(element_type et): m_et(et) {}
-				virtual ~NumberType() = default;
-
-				friend bool operator==(const NumberType& first, const NumberType& second);
-				friend bool operator!=(const NumberType& first, const NumberType& second);
-
-				virtual bool is_signed() const = 0;
-				virtual bool is_real() const = 0;
-				bool is_integral() const;
-
-				virtual bool compatible(const NumberType& other) const = 0;
-				virtual std::string to_string() const = 0;
-				virtual size_t size() const = 0;
-				element_type type() const { return m_et; }
-				operator element_type() const { return m_et; }
-			private:
-				const element_type m_et;
+		public:
```

Comment:
"All your base are belong to us"

Reply:
sorry, clang-format all the things.. I figured since we are copy pasting to a new repo it doesn't matter who owns the lines? I can revert these files if you prefer :)

Reply:
ahahhaha dont worry all ok! :) just had to put that oldschool quote there :P